hk2-api/src/main/java/org/glassfish/hk2/utilities/ClasspathDescriptorFileFinder.java

```diff
@@ -92,9 +96,38 @@ public ClasspathDescriptorFileFinder (ClassLoader cl) {
      * search for in the META-INF/hk2-locator directory
      */
     public ClasspathDescriptorFileFinder (ClassLoader cl, String... names) {
+        this.resourceBase = RESOURCE_BASE;
         this.classLoader = cl;
         this.names = names;
     }
+    
+    /**
+     * This constructor can be used to select the particular classloader
+     * to search for all files from the given locator directory. By using 
+     * this constructor, all files in the given directory are assumed to 
+     * be HK2 descriptor files.
+     *  
+     * @param locatorDirectory The directory of locator files, the class loader will search from 
+     * @param cl May not be null and must be the classloader to use when
+     * searching for HK2 descriptor files
+     */
+    public ClasspathDescriptorFileFinder (String locatorDirectory, ClassLoader cl) {
+        this.resourceBase = locatorDirectory.endsWith(File.separator) ? locatorDirectory : locatorDirectory + File.separatorChar;
+        this.classLoader = cl;
+        this.names = getLocatorFileNames(locatorDirectory);
+    }
+    private String[] getLocatorFileNames(String locatorDirectory) {
```

Comment:
Can you move the private method to the bottom of the file

Reply:
The private method is removed now :)

Reply:
I updated the code still in the same commit. Should I update it in another commit?